### PR TITLE
avoid error being logged when checksums.json is not found

### LINKED ISSUE #4301: `checksums.json` support clutters log with false crash messages
Since #3749 introduced support for `checksums.json` the log will now contain the following highly misleading error on almost every build:

> ERROR EasyBuild crashed with an error (at easybuild/base/exceptions.py:126 in __init__): Couldn't find file checksums.json anywhere, and downloading it is disabled... Paths attempted (in order): ...

I recently got this from another admin who is debugging a failure and thought this is the first error, which is understandable given the message "EasyBuild crashed with an error" which is clearly wrong at this point: The failure is very much expected.   
So `EasyBuildError` should not log anything **on construction** because it might be caught invalidating the logged message.

I found [`sys.excepthook`](https://stackoverflow.com/questions/6234405/logging-uncaught-exceptions-in-python) which is called for **unhandled** exceptions which is likely a better place for logging the exception then logging every (such) exception on creation even if it is caught and handled as part of an expected code path.

Alternative: Let `obtain_file` return `None` in case the file was not found, maybe based on a parameter `expect_success` for backwards compatibility.

CC @mboisson

### PR DESCRIPTION
**Pull request by PetrKralCZ**
(created using `eb --new-pr`)

edit: fixes #4301

### CODE REVIEW

**Review by boegel — CHANGES_REQUESTED** [review]
@PetrKralCZ As discussed:

- should enhance `test_obtain_file` in `test/framework/easyblock.py` to also test use of `warning_only` option;
    - to run only that specific test, use: `python3 -O -m test.framework.easyblock obtain_file`
- should enhance `test_checksum_step` to verify that log error doesn't happen anymore if there's no `checksums.json` file;

**Comment by Flamefire** [comment]
I don't see what it does to address #4254 which is rather addressed with #4150

As far as I can tell this only handles the case where the `checksums.json` file is missing, i.e. it fixes #4301

**Comment by boegel** [comment]
> I don't see what it does to address #4254 which is rather addressed with #4150
> 
> As far as I can tell this only handles the case where the `checksums.json` file is missing, i.e. it fixes #4301

You're right of course, thanks for the correction!

**Review by boegel — APPROVED** [review]
good to go now that a test that catches the issue is in place

File: test/framework/easyblock.py [hunk 1]
```diff
@@ -2424,6 +2425,30 @@ def test_checksum_step(self):
...
+            eb.fetch_sources()
+            eb.checksum_step()
+
+            fancylogger.logToScreen(enable=False, stdout=True)
+            stdout = self.get_stdout()
+
+            # make sure there's no error logged for not finding checksums.json,
+            # see also https://github.com/easybuilders/easybuild-framework/issues/4301
+            regex = re.compile("ERROR .*Couldn't find file checksums.json anywhere", re.M)
+            regex.search(stdout)
```
Comment (Flamefire) [inline test/framework/easyblock.py:2449]: @boegel did you miss to remove this?

Comment (boegel) [reply test/framework/easyblock.py:2449]: Ah, yes, that's not supposed to be there anymore...

Cleaning that up in #4345

**Pull request merged**